src/components/Scaler/index.js

```diff
@@ -26,201 +46,262 @@ class Scaler extends Component {
+      // When timeDomain changes, we need to update everything downstream.
+      const domainsByItemId = { ...this.state.domainsByItemId };
+      Object.keys(domainsByItemId).forEach(itemId => {
+        domainsByItemId[itemId][Axes.time] = this.props.dataContext.timeDomain;
       });
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ domainsByItemId });
     }
 
-    if (
-      Object.keys(domainUpdate).length ||
-      Object.keys(transformUpdate).length
-    ) {
+    if (!isEqual(prevProps.dataContext.series, this.props.dataContext.series)) {
+      const domainsByItemId = {};
+      const subDomainsByItemId = {};
+      []
+        .concat(this.props.dataContext.series)
+        .concat(this.props.dataContext.collections)
+        .forEach(item => {
+          domainsByItemId[item.id] = {
+            ...this.state.domainsByItemId[item.id],
+            [Axes.x]: [
+              ...(item.xDomain || Axes.x(this.state.domainsByItemId[item.id])),
+            ],
+            [Axes.y]: [
+              ...(item.yDomain || Axes.y(this.state.domainsByItemId[item.id])),
+            ],
+          };
+          subDomainsByItemId[item.id] = {
+            ...this.state.subDomainsByItemId[item.id],
+            [Axes.x]: [
+              ...(item.xSubDomain ||
+                Axes.x(this.state.subDomainsByItemId[item.id])),
+            ],
+            [Axes.y]: [
+              ...(item.ySubDomain ||
+                Axes.y(this.state.subDomainsByItemId[item.id])),
+            ],
+          };
+        });
       // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({
-        yTransformations: {
-          ...this.state.yTransformations,
-          ...transformUpdate,
-        },
-        ySubDomains: {
-          ...this.state.ySubDomains,
-          ...domainUpdate,
-        },
-      });
+      this.setState({ subDomainsByItemId, domainsByItemId });
     }
 
     // Handle changes in the base domain of the DataProvider
     const {
-      dataContext: {
-        xDomain: nextPropsDomain,
-        externalXDomain: nextExternalXDomain,
-      },
+      dataContext: { timeDomain: nextTimeDomain },
     } = this.props;
     const {
-      dataContext: {
-        xDomain: prevXDomain,
-        externalXDomain: prevExternalXDomain,
-      },
+      dataContext: { timeDomain: prevTimeDomain },
     } = prevProps;
-    if (!isEqual(prevExternalXDomain, nextExternalXDomain)) {
-      // External base domain changed (props on DataProvider)
-      // Reset state
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({
-        xSubDomain: nextExternalXDomain,
-        ySubDomains: {},
-        yTransformations: {},
-      });
-      return;
-    }
+
     if (
-      nextPropsDomain[0] !== prevXDomain[0] ||
-      nextPropsDomain[1] !== prevXDomain[1]
+      nextTimeDomain[0] !== prevTimeDomain[0] ||
+      nextTimeDomain[1] !== prevTimeDomain[1]
```

Comment:
isn't there already a check to see if the previous time domain is different from the new one? can't we just move the logic inside this if statement into the other one?

Reply:
Right you are! Done